src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -489,6 +489,13 @@ public static int onItemPickup(EntityItem entityItem, EntityPlayer player)
         return event.getResult() == Result.ALLOW ? 1 : 0;
     }
 
+    public static boolean onCanTakeItem(EntityPlayer player, ItemStack itemStack, boolean vanillaResult)
+    {
+        Event event = new net.minecraftforge.event.entity.player.CanTakeItemEvent(itemStack, player, vanillaResult);
+        MinecraftForge.EVENT_BUS.post(event);
+        return !event.isCanceled() && event.getResult() != Result.DENY;
```

Comment:
I think this event should only use `Result` and not canceled, so that `Result.DEFAULT` means use `vanillaResult`, that way implementations like in `SlotItemHandler` are still correct.
With the way it's implemented now, `SlotItemHandler` will allow taking items by default.

Reply:
So just remove the `!event.isCanceled() && ` part? or did I understand that wrong

Reply:
we need 3 cases:
* `Result.DENY`: return `false`
* `Result.DEFAULT`: return `vanillaResult`
* `Result.ALLOW`: return `true`

If you look at `SlotItemHandler` you can see why. Imagine nobody is listening to the event, this code is changing the method's behavior unless the `DEFAULT` action returns `vanillaResult`

Canceled is not really needed here as a way of changing the result, it's just to keep other mods from altering the result later.

Reply:
I see... So should I implement these all on one line? or have an if statement with 3 separate returns.

Reply:
Forge files like `ForgeEventFactory` aren't patches so they do not have line restrictions, so you can write it in a way that is more readable.

Reply:
Okay I did it using if statements for readability. Also, I put `return vanillaResult` as the default after the other cases.